flink-java/src/main/java/org/apache/flink/api/java/utils/RequiredParameters.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.java.utils;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Facility to manage required parameters in user defined functions.
+ */
+public class RequiredParameters {
+
+	private static final String HELP_TEXT_PARAM_DELIMITER = "\t";
+	private static final String HELP_TEXT_LINE_DELIMITER = "\n";
+
+	private HashMap<String, Option> data;
+
+	public RequiredParameters() {
+		this.data = new HashMap<>();
+	}
+
+	public Option add(String name) throws RequiredParametersException {
+		if (!this.data.containsKey(name)) {
+			Option option = new Option(name);
+			this.data.put(name, option);
+			return option;
+		} else {
+			throw new RequiredParametersException("Option with passed key already exists. " +  name);
+		}
+	}
+
+	public void add(Option option) throws RequiredParametersException {
+		if (!this.data.containsKey(option.getName())) {
+			this.data.put(option.getName(), option);
+		} else {
+			throw new RequiredParametersException("Option with passed key already exists. " +  option.getName());
+		}
+	}
+
+	/**
+	 * Check for all parameters in the parameterTool passed to the function:
+	 * - has a value been passed
+	 *   - if not, does the parameter have an associated default value
+	 * - does the type of the parameter match the one defined in RequiredParameters
+	 * - does the value provided in the parameterTool adhere to the choices defined in the option
+	 *
+	 * For any check which is not passed, a RequiredParametersException is thrown
+	 *
+	 * @param parameterTool - parameters supplied by the user.
+	 */
+	public void applyTo(ParameterTool parameterTool) throws RequiredParametersException {
```

Comment:
I think since the exception is an important part of the functionality, can you add a ` @throws` Javadoc comment to the method?